crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -504,8 +452,59 @@ impl<T: WasmInstance> ModuleInstance for WasmModuleInstance<T> {
 }
 
 impl<T: WasmInstance> WasmModuleInstance<T> {
+    fn call_reducer_internal(
```

Comment:
Could you add a doc comment describing the responsibilities of this method? In particular, I think it's non-obvious that it commits the transaction on success or cancels it on failure.

Reply:
c89b358816c116a729f4a0c1096c3ed581cb68ca

I didn't author the code, only moved it around, so this basically reflects my understanding from just reading the code. Also left a TODO regarding why it is that we can't get back our transaction easily.